### ISSUE TITLE
[code] Tags don't preserve whitespace in 1.8.12

### ISSUE THREAD

**Issue by Mako88** [mybb/mybb#2849]
In addition, if a custom css rule is used to set `white-space: pre`, every newline is doubled. (This is due to the nl2br function which should be edited to not run on <code> tags).

**Comment by Mako88** [comment]
Related to this, code tags replace tabs with spaces in function `mycode_parse_code`. This shouldn't be done since some coding languages (as well as patches) require tabs to be tab characters.

**Comment by arcalys** [comment]
Feel free to test the potential fix in the PR — feedback is welcome.

**Comment by arcalys** [comment]
To be added to 1.8.40 @dvz @Sama34?

**Comment by Sama34** [comment]
Absolutely, maybe by 2026 (🤪)

**Comment by arcalys** [comment]
> Absolutely, maybe by 2026 (🤪)

Nah, plenty of issues left to fix but I'm pretty sure we can get something available soon !
